engine/dlib/src/dlib/log.cpp

```diff
@@ -365,20 +409,22 @@ static void dmLogThread(void* args)
         // Currently no support for that and hence the sleep here
         dmTime::Sleep(1000 * 30);
         dmLogUpdateNetwork();
-        dmMessage::Dispatch(self->m_MessageSocket, dmLogDispatch, (void*) &run);
+        dmMessage::Dispatch(server->m_MessageSocket, dmLogDispatch, (void*) &run);
     }
 }
 
 void LogInitialize(const LogParams* params)
 {
     g_TotalBytesLogged = 0;
 
-    if (g_dmLogServer)
+    if (dmAtomicGet32(&g_LogServerInitialized) != 0)
     {
         fprintf(stderr, "ERROR:DLIB: dmLog already initialized\n");
         return;
     }
 
+    dmSpinlock::Init(&g_LogServerLock);
```

Comment:
...and a spin lock to make sure no one is logging/shutting down the log system at the same time.